gooddata-java/src/test/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigServiceTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.service.hierarchicalconfig;
+
+import com.gooddata.sdk.common.GoodDataException;
+import com.gooddata.sdk.model.hierarchicalconfig.ConfigItem;
+import com.gooddata.sdk.model.hierarchicalconfig.ConfigItems;
+import com.gooddata.sdk.model.project.Project;
+import com.gooddata.sdk.service.GoodDataSettings;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class HierarchicalConfigServiceTest {
+
+    private static final String PROJECT_ID = "11";
+    private static final String SETTING_ITEM_NAME = "item1";
+    private static final String PROJECT_SETTINGS_URI = "/gdc/projects/11/config";
+    private static final String PROJECT_SETTING_ITEM_URI = "/gdc/projects/11/config/" + SETTING_ITEM_NAME;
+
+    @Mock
+    private Project project;
+    @Mock
+    private ConfigItem configItem;
+    @Mock
+    private RestTemplate restTemplate;
+
+    private HierarchicalConfigService hierarchicalConfigService;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this).close();
+        hierarchicalConfigService = new HierarchicalConfigService(restTemplate, new GoodDataSettings());
+        when(project.getId()).thenReturn(PROJECT_ID);
+    }
+
+    /*--- Project Settings ---*/
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void whenNullArgThenListProjectSettingsShouldThrow() {
```

Comment:
Still uses old "setting" naming. Please grep whole PR for this name and change everywhere where it makes sense

Reply:
Changed